Shymansky Pavel

pavl1n opened the pull request:
Description: There is a problem with the output of information in printArticles() function. Also few issues connected with argparse module, unrealised json format fully. And russian is not supported here at moment.

pavl1n committed:
First iteration

Description: There is a problem with the output of information in printArticles() function. Also few issues connected with argparse module, unrealised json format fully. And russian is not supported here at moment.

AlexeiBuzuma commented on .idea/PythonHomework.iml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="PYTHON_MODULE" version="4">
```
Как я уже писал в чате: 
> Мета-файлы вашего текстового редактора в репозитории не должны хранится. Если вы используете PyCharm, то вам стоит добавить .idea/ в .gitignore файл.

pavl1n committed:
Delete misc.xml

pavl1n committed:
Delete .gitignore

pavl1n committed:
Delete profiles_settings.xml

pavl1n committed:
Delete PythonHomework.iml

pavl1n committed:
Delete modules.xml

pavl1n committed:
Delete vcs.xml

pavl1n committed:
Second iteration

In this version was fixed bugs with output and added distribution feature

pavl1n committed:
Merge branch 'FinalTask' of https://github.com/pavl1n/PythonHomework into FinalTask

pavl1n committed:
Second iteration updated

In this version was added such feautures as unittest, some logs, saving news in 'json' format, and fixed bugs related with output

pavl1n committed:
Delete cache.py

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,15 @@
+    description = 'CLI utility to process RSS',
+    author = 'Pavel Shymansky',
+    py_module = ['rss.parser','arg.py'],
+    install_requires = ['feedparser', 'bs4'], #!!!!!!!!!!!!!!!!!!!не забытЬ!!!!!!!!!!!!!
```
Веселый коментарий :) 
Обычно для подобного рода комментариев для "незабывания" используют " # ToDo: " 
многие среды разработки умеют красиво их подчеркивать

AlexeiBuzuma commented on rss_parser.py:
```diff
@@ -0,0 +1,115 @@
+    logger.info('Limit is {}'.format(args.limit))
+
+
+def versionInfo(args):
```
название метода должно быть маленькими буквами через нижнее подчеркивание

AlexeiBuzuma commented on loggerfile.py:
```diff
@@ -0,0 +1,12 @@
+import logging
+extra = {'app_name':'rss_parser'}
+
+logger = logging.getLogger(__name__)
```
в глобальной области видимости обычно не выполняют никакой логики. Лучше перенести эту логику в отдельную функцию, а потом ее вызвать из какой-нибудь функции (например main, или что-то в этом роде)

AlexeiBuzuma commented on arg.py:
```diff
@@ -0,0 +1,10 @@
+import argparse
+
+parser = argparse.ArgumentParser(description='Pure python command-line RSS reader')
```
Такая же ситуация, нужно перенести эту логику в отдельную функцию

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,15 @@
+    python_requires='>=3.8',
+    entry_points ='''
+    [console_scripts]
+    rss-parser=rss_parser: main
```
1) в тех задании описано, что утилита должна называться `rss-reader`
2) не работает после установки:
```bash 
# rss-parser --help
Traceback (most recent call last):
  File "/usr/local/bin/rss-parser", line 5, in <module>
    from rss_parser import main
ModuleNotFoundError: No module named 'rss_parser'
```

pavl1n committed:
All comments was fixed

Also was added feature --date which cache news, but now it not fully implemented. Method --version was moved in separate file. Logs was added to the another functions

pavl1n committed:
All comments was fixed

Also was added feature --date which cache news, but now it not fully implemented. Method --version was moved in separate file. Logs was added to the another functions

pavl1n committed:
Last commit

Add some more logs, add to_pdf function but not implemented fully, add requirements.txt, README.md

AlexeiBuzuma commented on arg.py:
```diff
@@ -0,0 +1,14 @@
+    parser.add_argument("--verbose", action='store_true', help='Outputs verbose status messages')
+    parser.add_argument("--limit", help='Limit news topics', type=int)
+    parser.add_argument("--date", help='Shows cached news on introduced day', type=int)
+    parser.add_argument("--to_pdf",action='store_true', help = 'Converts news into PDF format' )
```
Конвертация сделана только в один формат?

AlexeiBuzuma commented:
К сожалению, не работает из-за проблем с импортом 
```bash
# rss-reader --help
Traceback (most recent call last):
  File "/usr/local/bin/rss-reader", line 5, in <module>
    from rss_reader import main
ModuleNotFoundError: No module named 'rss_reader'
```

pavl1n commented:
@AlexeiBuzuma а можешь пожалуйста подсказать как вы проверяли работы? Потому что я чекал на чистой винде и голом линуксе и всё работало

AlexeiBuzuma commented:
Работы проверялись в докер контейнерах python:latest.
В чистом контейнере клонировался репозиторий с кодом и производилась установка через  `pip install .` 
более подробно этот процесс описывался в нашем чате в слаке 

Пункты, которые могут быть улучшены: 
1) файл date_converter.py. Стандартная библиотека пайтона содержит в себе много готовых вещей. Например есть модуль datetime, который возволит сделать то, что реализованно в функции convert_date
2) не все имена переменных соответствуют pep8 
3) Нет использования ООП, хотя эта парадигма очень уместна для данной задачи 
4) функционально для verbose может быть реализована с помощью функциональности из модуля logging 
5) функция parse вызывается в приложении 8 раз, каждый раз при вызове этой функции отправляется запрос на получения XML с новостями. Это достаточно не эффективно, плюс во время работы приложения новости могут измениться и на последний запрос, например, данный скрипт может работать уже с отличными данными от первоначальных новостей.
